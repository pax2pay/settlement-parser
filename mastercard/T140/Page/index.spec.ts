import "jest"
import { Page } from "."

describe("mastercard.T140.Page", () => {
	const pages = [
		[
			"1IP727010-AA                                           MASTERCARD WORLDWIDE                                       RUN DATE: 12/21/23 ",
			"                                                CLEARING CYCLE 003 - ACKNOWLEDGEMENT                              PAGE NO:         1 ",
			"                                                              2023-12-21                                                             ",
			" MEMBER ID: 00000033400                                                                                                              ",
			"                                                         NO DATA TO REPORT                                                           ",
		].join("\n"),
		[
			"1IP727020-AA                                           MASTERCARD WORLDWIDE                                       RUN DATE: 12/21/23 ",
			" ACCEPTANCE BRAND: MCC                          CLEARING CYCLE 003 - NOTIFICATION                                 PAGE NO:         1 ",
			" BUSINESS SERVICE LEVEL: MBR-TO-MBR AGREEMENT                 2023-12-21                                                             ",
			" BUSINESS SERVICE ID: 000900                                                                                                         ",
			" FILE ID: 001/231221/00000004882/03402                                                                                               ",
			" MEMBER ID: 00000033400                                                                                                              ",
			"                                                                                                                                     ",
			" MASTERCARD SETTLED                                                     RECON                         FEE                            ",
			"                                                                         CURR                         CURR                           ",
			" TRANS. FUNC. PROC.CODE       IRD   COUNTS               RECON AMOUNT    CODE       TRANS FEE         CODE                           ",
			" ------------ --------------- --- -------- -------------------------- ------- ----------------------- -------                        ",
			" FIRST PRES.  PURCHASE   ORIG BB         1                    0.73 DR 826-GBP                 0.02 CR 826-GBP                        ",
			" ------------ --------------- --- -------- -------------------------- ------- ----------------------- -------                        ",
			" FIRST PRES.  TOTAL                      1                    0.73 DR 826-GBP                 0.02 CR 826-GBP                        ",
			"                                                                                                                                     ",
			" MBR-TO-MBR AGREEMENT                                                                                                                ",
			" MASTERCARD SETTLED                                                                                                                  ",
			" BUSINESS SERVICE ID SUBTOTAL            1                    0.73 DR 826-GBP                 0.02 CR 826-GBP                        ",
		].join("\n"),
		[
			"1IP727020-AA                                           MASTERCARD WORLDWIDE                                       RUN DATE: 12/21/23 ",
			"                                                CLEARING CYCLE 003 - NOTIFICATION                                 PAGE NO:         1 ",
			"                                                              2023-12-21                                                             ",
			"                                                                                                                                     ",
			" ACCEPTANCE BRAND : MCC                                                                                                              ",
			" BUSINESS SERVICE LEVEL :MBR-TO-MBR AGREE                                                                                            ",
			" MEMBER ID: 00000033400                                                                                                              ",
			" CURRENCY CODE : 826-GBP                                                                                                             ",
			"                                                                                                                                     ",
			" BUSINESS                                                                                                                            ",
			" SERVICE                               ORIG/                                                                                         ",
			"     ID            FILE ID             RVSL          RECON. AMOUNT                   TRANSACTION FEE                                 ",
			" -------- ---------------------------- ----        -------------------              -------------------                              ",
			" 000900   001/231221/00000004882/03402 ORIG                     0.73 DR                         0.02 CR                              ",
			"                                                                                                                                     ",
			"                             GRAND TOTAL                        0.73 DR                         0.02 CR                              ",
			"                                                                                                                                     ",
		].join("\n"),
	]

	it("parse acknowledgement", () => expect(Page.parse(pages[0])).toMatchSnapshot())
	it("parse notification", () => expect(Page.parse(pages[1])).toMatchSnapshot())
	it("parse notification", () => expect(Page.parse(pages[2])).toMatchSnapshot())
})
