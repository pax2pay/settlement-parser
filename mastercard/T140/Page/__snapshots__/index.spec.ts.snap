// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mastercard.T140.Page parse acknowledgement 1`] = `
Object {
  "body": Array [],
  "header": Object {
    "cycle": 3,
    "date": "2023-12-21",
    "member": 33400,
    "number": 1,
    "runDate": "2023-12-21",
    "type": "acknowledgement",
  },
}
`;

exports[`mastercard.T140.Page parse notification 1`] = `
Object {
  "body": Array [
    Object {
      "content": Array [
        Array [
          "FIRST PRES.",
          "PURCHASE   ORIG",
          "BB",
          "1",
          "0.73 DR",
          "826-GBP",
          "0.02 CR",
          "826-GBP",
        ],
      ],
      "footers": Array [
        "FIRST PRES.",
        "TOTAL",
        "",
        "1",
        "0.73 DR",
        "826-GBP",
        "0.02 CR",
        "826-GBP",
      ],
      "headers": Array [
        "TRANS. FUNC.",
        "PROC.CODE",
        "IRD",
        "COUNTS",
        "RECON AMOUNT",
        "RECON CURR CODE",
        "TRANS FEE",
        "FEE CURR CODE",
      ],
    },
  ],
  "header": Object {
    "brand": "MCC",
    "cycle": 3,
    "date": "2023-12-21",
    "file": "001/231221/00000004882/03402",
    "level": "mbr-to-mbr agreement",
    "member": 33400,
    "number": 1,
    "runDate": "2023-12-21",
    "type": "notification",
  },
}
`;

exports[`mastercard.T140.Page parse notification 2`] = `
Object {
  "body": Array [
    Object {
      "header": Object {
        "brand": "MCC",
        "currency": "GBP",
        "level": "mbr-to-mbr agreement",
        "member": 33400,
      },
      "table": Object {
        "content": Array [
          Array [
            "000900",
            "001/231221/00000004882/03402",
            "ORIG",
            "0.73 DR",
            "0.02 CR",
          ],
        ],
        "footers": undefined,
        "headers": Array [
          "BUSINESS SERVICE ID",
          "FILE ID",
          "ORIG/ RVSL",
          "RECON. AMOUNT",
          "TRANSACTION FEE",
        ],
      },
    },
  ],
  "header": Object {
    "cycle": 3,
    "date": "2023-12-21",
    "number": 1,
    "runDate": "2023-12-21",
    "type": "notification",
  },
}
`;
